src/nvim/window.c

```diff
@@ -6248,7 +6366,7 @@ void win_comp_scroll(win_T *wp)
 {
   const long old_w_p_scr = wp->w_p_scr;
 
-  wp->w_p_scr = wp->w_height / 2;
+  wp->w_p_scr = wp->w_height_inner / 2;
```

Comment:
The options code look fine. `if_false` only has effect when `condition` is present.

Reply:
> So this fixes the splitscroll test

It appears it still fails somewhere on CI, couldn't reproduce it locally anymore.

Reply:
> I suspect it has something to do with default option values.

Right, we have to unset `winbar` as it is a global option for nvim. As opposed to vim's window local winbar. Fixed in latest commit.

(I'm aware we can `setlocal winbar` but this would result in the winbar disappearing as soon as the window is closed, which happens a lot in this test. As such, we are not really testing the same window layouts as in vim, as each window will have a winbar. But at least we are confirming that no scrolling happens with winbar enabled.)